File: legado/legado.py
import json
import os

arquivo = "db.json"

if os.path.exists(arquivo):
    try:
        f = open(arquivo, "r", encoding="utf-8")
        dados = json.load(f)
        f.close()
    except:
        dados = []
else:
    dados = []

while True:
    print("\n=== SISTEMA DE CADASTRO LEGADO ===")
    print("1 - Cadastrar novo")
    print("2 - Sair")
    opcao = input("Opção: ")

    if opcao == "1":
        print("\n--- Novo Cadastro ---")
        nome = input("Nome: ")
        idade = input("Idade: ")
        email = input("Email: ")

        registro = {
            "nome": nome,
            "idade": idade,
            "email": email
        }

        dados.append(registro)

        try:
            f = open(arquivo, "w", encoding="utf-8")
            json.dump(dados, f, indent=4, ensure_ascii=False)
            f.close()
            print("\nCadastro salvo!")
        except:
            print("Erro ao salvar (mas o programa continua)")

    elif opcao == "2":
        print("Saindo...")
        break

    else:
        print("Opção inválida. Tente de novo.")

File: legado/legado.js
const fs = require("fs");

let arquivo = "db.json";
let dados = [];

// tenta carregar "do jeito que der"
if (fs.existsSync(arquivo)) {
    try {
        let conteudo = fs.readFileSync(arquivo).toString();
        dados = JSON.parse(conteudo);
    } catch (e) {
        dados = [];
    }
} else {
    dados = [];
}

// input sujo via readline-sync
const rl = require("readline-sync");

while (true) {
    console.log("\n=== SISTEMA DE CADASTRO LEGADO (JS) ===");
    console.log("1 - Cadastrar novo");
    console.log("2 - Sair");

    let opcao = rl.question("Opcao: ");

    if (opcao == "1") {
        console.log("\n--- Novo Cadastro ---");
        let nome = rl.question("Nome: ");
        let idade = rl.question("Idade: ");
        let email = rl.question("Email: ");

        let registro = {
            nome: nome,
            idade: idade,
            email: email
        };

        dados.push(registro);

        try {
            fs.writeFileSync(arquivo, JSON.stringify(dados, null, 4));
            console.log("\nCadastro salvo!");
        } catch (e) {
            console.log("Erro ao salvar, mas vida que segue.");
        }
    }

    else if (opcao == "2") {
        console.log("Saindo...");
        break;
    }

    else {
        console.log("Opcao invalida.");
    }
}
